source/opt/desc_sroa.h

```diff
@@ -32,7 +32,9 @@ namespace opt {
 // Documented in optimizer.hpp
 class DescriptorScalarReplacement : public Pass {
  public:
-  DescriptorScalarReplacement() {}
+  DescriptorScalarReplacement(bool flatten_composites, bool flatten_arrays)
+      : flatten_composites_(flatten_composites),
+        flatten_arrays_(flatten_arrays) {}
 
   const char* name() const override { return "descriptor-scalar-replacement"; }
```

Comment:
This is suppose to match the command line option used to rerun this pass. To debug, someone can use `--print-all`, and get the spir-v before the pass with the option needed to rerun the next pass. This might need to look at the new member variables, and return a different name based on which options are used.